pkg/apis/internalversion/metric_type.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internalversion
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Metric provides metrics configuration.
+type Metric struct {
+	//+k8s:conversion-gen=false
+	metav1.TypeMeta `json:",inline"`
```

Comment:
The `metadata` is not in

```
metav1.TypeMeta `json:",inline"`
```

that is in 

```
metav1.ObjectMeta `json:"metadata"`
```

Is it right?

Reply:
Exactly! It will be embedded on default serialization like:
`{"name": "fake-node", ...}`, but we really want `{"metadata":{"name":"fake-node"}}`

Reply:
Ohh, I see, our understanding of this is a little off,

I meant that `{{ .metadata.name }}` is filled with `corev1.Node` and not `internalversion.Metric`.

Reply:
Oh, I see, but the problem persists for the `path`:
```yaml
path: "/metrics/nodes/{{ .metadata.name }}"
```
My understanding was to create path by executing a template. At this point, we don't have a node yet. We can eliminate the need for `json` if the path is not dynamic like this, but rather a constant. That way, we can always use `corev1.Node` which does have `json` struct tags

Reply:
Oh then I misunderstood the spec for:
```yaml
name: node-name
```
So the name of the metric has nothing to do with the node it targets? If that is the case, I can change the server implementation to search for the `{{ }}` in the URL and treat it as a dynamic parameter (replace it with `/metrics/nodes/{node-name}`). It is just going to be slightly more complicated if there are multiple field selectors like `{{ .metadata.namespace }}`. Then this information would need to be propagated to the `CustomRegistry` so it knows how to fetch the correct registry for a specific endpoint. I just need to figure out how to propagate that metadata to the registry, because the metrics updates are happening per node, while the selector can have a brother scope (in this case, the namespace)

Reply:
Great :smile: on it!